src/convertToObject.js

```diff
@@ -6,7 +6,19 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const arrayOfSource = sourceString
+    .split(';')
+    .filter((part) => part.trim() !== '');
+  const MappedArrayOfSource = arrayOfSource.map((part) => {
+    return part.split(':');
+  });
```

Comment:
The variable 'MappedArrayOfSource' does not follow JavaScript naming conventions. Variable names should be in camelCase. Consider renaming it to 'mappedArrayOfSource'.